### PR TITLE
Add post 2.10 remote store tweaks

### LINKED ISSUE #5102: [DOC] Update segment replication consideration section
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
1. Segment replication consideration section is broken as `Read-after-write guarantees` is merged onto point 4. This is correct in 2.8.x [documentation](https://opensearch.org/docs/2.8/tuning-your-cluster/availability-and-recovery/segment-replication/index/#considerations)
```
4. Segment replication leads to increased network congestion on primary shards using node-to-node replication because replica shards fetch updates from the primary shard. With remote-backed storage, the primary shard can upload segments to, and the replicas can fetch updates from, the remote-backed storage. This helps offload responsibilities from the primary shard to the remote-backed storage. Read-after-write guarantees: Segment replication does not currently support setting the refresh policy to wait_for. If you set the refresh query parameter to wait_for and then ingest documents, you’ll [get](https://opensearch.org/docs/2.10/api-reference/document-apis/get-documents/) a response only after the primary node has refreshed and made those documents searchable. Replica shards will respond only after having written to their local translog. If real-time reads are needed, consider using the get or [mget](https://opensearch.org/docs/2.10/api-reference/document-apis/multi-get/) API operations.
```
Correct
```
4. Segment replication leads to increased network congestion on primary shards using node-to-node replication because replica shards fetch updates from the primary shard. With remote-backed storage, the primary shard can upload segments to, and the replicas can fetch updates from, the remote-backed storage. This helps offload responsibilities from the primary shard to the remote-backed storage. 
5. Read-after-write guarantees: Segment replication does not currently support setting the refresh policy to wait_for. If you set the refresh query parameter to wait_for and then ingest documents, you’ll [get](https://opensearch.org/docs/2.10/api-reference/document-apis/get-documents/) a response only after the primary node has refreshed and made those documents searchable. Replica shards will respond only after having written to their local translog. If real-time reads are needed, consider using the get or [mget](https://opensearch.org/docs/2.10/api-reference/document-apis/multi-get/) API operations.
```

2. Read-after-write guarantees is also not supported when using `refresh` param as `true` during[ indexing operation](https://opensearch.org/docs/2.10/api-reference/document-apis/index-document/#url-parameters). We need to update point 5 above and include this limitation as well. 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
Related PR: https://github.com/opensearch-project/documentation-website/pull/5037

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
- Fixes #5102 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ashking94 — APPROVED** [review]
LGTM the section on remote store main page.

File: _tuning-your-cluster/availability-and-recovery/remote-store/snapshot-interoperability.md [hunk 1]
```diff
@@ -8,21 +8,24 @@ grand_parent: Availability and recovery
...
+Shallow copy snapshots allow you to reference data from an entire remote-backed repository instead of storing all of the data from the segment in a snapshot repository. This makes accessing segment data faster than using normal snapshots because segment data is not stored in the snapshot repository.
 
 ## Enabling shallow snapshots
 
-Use the [Cluster Settings API]({{site.url}}{{site.baseurl}}/api-reference/cluster-api/cluster-settings/) to enable the `remote_store_index_shallow_copy` repository setting, as shown in the following example:
+Use the [Snapshot API]({{site.url}}{{site.baseurl}}/api-reference/snapshots/create-repository/) and set the `remote_store_index_shallow_copy` repository setting to `true` to enable shallow snapshot copies, as shown in the following example:
 
 ```bash
-PUT _cluster/settings
+curl -X PUT "localhost:9200/_snapshot/snap_repo?pretty" -H 'Content-Type: application/json' -d'
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/remote-store/snapshot-interoperability.md:18]:
```suggestion
PUT /_snapshot/snap_repo
```

File: _tuning-your-cluster/availability-and-recovery/remote-store/snapshot-interoperability.md [hunk 2]
```diff
@@ -8,21 +8,24 @@ grand_parent: Availability and recovery
...
-   }
-}
+        "type": "s3",
+        "settings": {
+            "bucket": "test-bucket",
+            "base_path": "daily-snaps",
+            "remote_store_index_shallow_copy": true
+        }
+    }
+'
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/remote-store/snapshot-interoperability.md:27]:
```suggestion
```

File: _tuning-your-cluster/availability-and-recovery/remote-store/snapshot-interoperability.md [hunk 3]
```diff
@@ -8,21 +8,24 @@ grand_parent: Availability and recovery
...
 ```
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/remote-store/snapshot-interoperability.md:28]:
```suggestion
```
{% include copy-curl.html %}
```

**Review by kolchfa-aws — APPROVED** [review]
For readability, we normally don't use curl and include a copy curl button that will format a request as curl.

**Pull request merged**